### ISSUE TITLE
Recreating "E: Novel contextual references" training dataset doubt.

### ISSUE THREAD

**Issue by aamacaya** [LauraRuis/multimodal_seq2seq_gSCAN#1]
Hi!

I am trying to recreate the "E: Novel contextual references" experiment there is something that I don't understand.

The paper says that:

> We investigate whether a model can grasp relativity in language by considering a
scenario where objects of a specific size (size 2) are never targets correctly picked by the ‘small’
modifier during training (see Figure 5).

My question is: On training, how do you get rid of the examples where objects of size 2 are targets correctly picked by the ‘small’ modifier?

Because, for what I understand from all_experiments.sh, when running:

```python3.7 -m seq2seq --mode=train --max_decoding_steps=120 --max_testing_examples=2000 --data_directory=data/compositional_splits --attention_type=bahdanau --no_auxiliary_task --conditional_attention --output_directory=adverb_k_1_run_3 --training_batch_size=200 --max_training_iterations=200000 --seed=66```

it will train de model with the "train" split which contains those kinds of examples.

What would be the equivalent command for training the "E: Novel contextual references" experiment? (so I can get the same results of the paper).

**Comment by LauraRuis** [comment]
Hi :) Cool that you're trying to recreate the experiment! 

The sentence from the paper you're referring to is indeed a bit ambiguous maybe. If you look at the sentence that follows it: 

" In other words, we hold out for testing all world states where the circle of size 2 is the target and the smallest
circle in the world, paired with an instruction containing the word ‘small’"

So the training set does indeed contain other object of size 2 that are called 'small', like cylinders or squares, but never circles that are of size 2 and called 'small'. You don't need to change anything about the data to recreate this experiment. If you use the command from `all_experiments.sh` it will train on the training set without these examples, and then you can test this trained models on split E:

```
python3.7 -m seq2seq --mode=test --data_directory=<path/to/folder/with/dataset.txt/> --attention_type=bahdanau --no_auxiliary_task --conditional_attention --output_directory=<path/containing/trained/models> --resume_from_file=<path/containing/trained/models>/model_best.pth.tar --splits=situational_2 --output_file_name=predict.json --max_decoding_steps=120
```

Make sure to run your trained model also on the random test set (split A) to check whether it gives a high exact match there! Let me know if you have more questions :)

**Comment by aamacaya** [comment]
Thank's @LauraRuis!

Now everything is clear! I will continue recreating the experiment and trying new thigs.
